extensions/VisualEditor/wikia/ApiAddMediaTemporary.php

```diff
@@ -43,7 +43,7 @@ private function executeImage() {
 			// If wiki is Japanese content, then we check if anonymous edit is allowed. INT-158
 			// This condition will be changed as soon as Mercury has login for all wikis.
 			if ( $wgContLanguageCode === 'ja' ) {
-				if ( $wgDisableAnonymousEditing ) {
+				if ( $wgDisableAnonymousEditing && !$this->mUser->isLoggedIn()) {
```

Comment:
Thanks, you are right. I checked all cases and confirmed that we want to skip checkPremission() when it is JA and anonymous editing is allowed. So negating that is when we want to do checkPermission(), as you wrote above.